src/symbolic/solver.ml

```diff
@@ -18,8 +18,9 @@ let check (S (solver_module, s)) pc =
   let module Solver = (val solver_module) in
   Solver.check_set s pc
 
-let model (S (solver_module, s)) ~symbols ~pc =
+let model (S (solver_module, s)) ~scoped_symbols ~pc =
   let module Solver = (val solver_module) in
+  let symbols = Some (Scoped_symbol.only_symbols scoped_symbols) in
```

Comment:
Something looks suspicious here: why did we use an option before and always have `Some` now? We should either remove this `Some` (and fix the following code) or change something elsewhere.